packages/headless/src/controllers/insight/search-box/headless-insight-search-box.ts

```diff
@@ -30,6 +29,42 @@ export interface SearchBoxProps {
   options?: SearchBoxOptions;
 }
 
+export interface SearchBox extends Controller {
+  /**
+   * Updates the search box text value and shows the suggestions for that value.
+   *
+   * @param value - The string value to update the search box with.
+   */
+  updateText(value: string): void;
+
+  /**
+   * Clears the search box text and the suggestions.
+   */
+  clear(): void;
+
+  /**
+   * Shows the suggestions for the current search box value.
+   */
+  showSuggestions(): void;
+
+  /**
+   * Selects a suggestion and calls `submit`.
+   *
+   * @param value - The string value of the suggestion to select
+   */
+  selectSuggestion(value: string): void;
+
+  /**
+   * Deselects all facets and triggers a search query.
+   */
+  submit(): void;
+
+  /**
+   * The state of the `SearchBox` controller.
+   */
+  state: SearchBoxState;
+}
+
```

Comment:
The insight use case was entirely exporting the core search box interface that has a different submit function. This is technically a breaking change but can be considered a bug. The other stuff was never supposed to be exposed...

Reply:
Headless insight panel is also "used by us" (Coveo R&D), not clients/partners directly.

So this change is completely fine as long as everything still works in e2e tests.